mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraBlobsDAO.java

```diff
@@ -0,0 +1,176 @@
+import java.io.SequenceInputStream;
+import java.nio.ByteBuffer;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+import javax.inject.Inject;
+
+import org.apache.james.backends.cassandra.utils.CassandraAsyncExecutor;
+import org.apache.james.backends.cassandra.utils.CassandraUtils;
+import org.apache.james.util.FluentFutureStream;
+import org.apache.commons.lang3.tuple.Pair;
+
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Row;
+import com.datastax.driver.core.Session;
+import com.datastax.driver.core.querybuilder.QueryBuilder;
+import com.datastax.driver.core.utils.UUIDs;
+import com.github.steveash.guavate.Guavate;
+import com.google.common.collect.ImmutableMap;
+
+public class CassandraBlobsDAO {
+
+    public static final int CHUNK_SIZE = 1024;
+    private final CassandraAsyncExecutor cassandraAsyncExecutor;
+    private final PreparedStatement insert;
+    private final PreparedStatement insertPart;
+    private final PreparedStatement delete;
+    private final PreparedStatement select;
+    private final PreparedStatement selectPart;
+
+    @Inject
+    public CassandraBlobsDAO(Session session) {
+        this.cassandraAsyncExecutor = new CassandraAsyncExecutor(session);
+        this.insert = prepareInsert(session);
+        this.delete = prepareDelete(session);
+        this.select = prepareSelect(session);
+
+        this.insertPart = prepareInsertPart(session);
+        this.selectPart = prepareSelectPart(session);
+    }
+
+    private PreparedStatement prepareSelect(Session session) {
+        return session.prepare(select()
+            .from(Blobs.TABLE_NAME)
+            .where(eq(Blobs.ID, bindMarker(Blobs.ID))));
+    }
+
+    private PreparedStatement prepareSelectPart(Session session) {
+        return session.prepare(select()
+                .from(BlobParts.TABLE_NAME)
+                .where(eq(BlobParts.ID, bindMarker(BlobParts.ID))));
+    }
+
+    private PreparedStatement prepareInsert(Session session) {
+        return session.prepare(insertInto(Blobs.TABLE_NAME)
+                .value(Blobs.ID, bindMarker(Blobs.ID))
+                .value(Blobs.POSITION, bindMarker(Blobs.POSITION))
+                .value(Blobs.PART, bindMarker(Blobs.PART)));
+    }
+
+    private PreparedStatement prepareInsertPart(Session session) {
+        return session.prepare(insertInto(BlobParts.TABLE_NAME)
+                .value(BlobParts.ID, bindMarker(BlobParts.ID))
+                .value(BlobParts.DATA, bindMarker(BlobParts.DATA)));
+    }
+
+    private PreparedStatement prepareDelete(Session session) {
+        return session.prepare(QueryBuilder.delete()
+                .from(Blobs.TABLE_NAME)
+                .where(eq(Blobs.ID, bindMarker(Blobs.ID))));
+    }
+
+    public CompletableFuture<Optional<UUID>> save(byte[] data) {
```

Comment:
After looking at the code, only CassandraMessageIdMapper deletes messages (as CassandraMessageMapper don't know if it is in another mailbox).

@mbaechler  deleting the references to the message is enough. It's what is done with Attachments, and it seems it is a problem to nobody. Why not doing it also with messages, and embrace similar benefits.

I will bring this discussion in the daily, so that the team can choose, and move fast.